streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -345,8 +343,17 @@ private SinkNodeFactory(final String name,
         }
     }
 
+    public void setTopologyName(final String namedTopology) {
```

Comment:
Could we move the logic of `namedTopology.setTopologyName(topologyName);` into the constructor of `NamedTopologyStreamsBuilder(final String topologyName)` itself, and then call the constructor of `NamedTopology` directly, so that we can still have a final field in line 138 above?

Reply:
I tried to, but just couldn't make it work. It has to do with Java and subclassing quirks like constructing the parent before the child. It seems to be pretty much impossible to set things up so that everything is `final` -- if we set the `topologyName` in the NamedTopology constructor, then it's not accessible (ie always null) when we call the `InternalTopologyBuilder`'s constructor since that occurs during the parent `Topology`'s construction.

It's definitely annoying, but at least we should be able to clean things up once we go through a KIP and don't need to subclass like this.

Reply:
I see.